core/src/main/scala/org/genivi/sota/core/db/Campaigns.scala

```diff
@@ -22,27 +23,36 @@ object Campaigns {
   import Campaign._
   import SotaCoreErrors._
 
+  private def optionPackageId(name: Option[PackageId.Name], version: Option[PackageId.Version]): Option[PackageId] =
+    (name, version) match {
+      case (Some(n), Some(v)) => Some(PackageId(n, v))
+      case _ => None
+    }
```

Comment:
this is `name.flatMap(n => version.map(v => (n, v))` or:
``
for {
  n <- name
  v <- version
} yield (n, v)
```